.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,28 @@
+name: Go
+
+on:
+  push:
```

Comment:
Shall we run it to all the branches? I'd think would be quite useful to know if the `main` branch is in good shape too.

Reply:
i believe we only have test and PR branches - do you think we should always run tests ?

Reply:
if so we can do this 
```
on:
  push:
    branches:
      - '**'
  pull_request:
    branches:
      - '**'
```

Reply:
> i believe we only have test and PR branches - do you think we should always run tests ?

I'm a bit on the yes side, but maybe @masa-finance/protocol have different thoughts in here?